WSLGd/FontMonitor.cpp

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+#include "FontMonitor.h"
+#include "common.h"
+
+#define USER_DISTRO_MOUNT_PATH "/mnt/wslg/distro"
+#define USER_DISTRO_FONTPATH USER_DISTRO_MOUNT_PATH "/usr/share/fonts/"
+
+constexpr auto c_fontsdir = "fonts.dir";
+constexpr auto c_xset = "/usr/bin/xset";
+
+wslgd::FontFolder::FontFolder(int fd, const char *path)
+{
+    LOG_INFO("FontMonitor: start monitoring %s", path);
+
+    m_fd.reset(fd);
+    m_path = path;
+
+    /* check if folder is already ready to be added to font path */
+    try {
+        std::string fonts_dir(path);
+        fonts_dir += c_fontsdir;
+        if (access(fonts_dir.c_str(), R_OK) == 0) {
+            ModifyX11FontPath(true);
+        }
+        /* add watch for install or uninstall of fonts on this folder */
+        THROW_LAST_ERROR_IF((m_wd = inotify_add_watch(m_fd.get(), path, IN_CREATE|IN_CLOSE_WRITE|IN_DELETE|IN_MOVED_TO|IN_MOVED_FROM)) < 0);
+    }
+    CATCH_LOG();
+}
+
+wslgd::FontFolder::~FontFolder()
+{
+    LOG_INFO("FontMonitor: stop monitoring %s", m_path.c_str());
+
+    ModifyX11FontPath(false);
+
+    /* if still under watch, remove it */
+    if (m_wd >= 0) {
+        inotify_rm_watch(m_fd.get(), m_wd);
+        m_wd = -1;
+    }
+
+    m_fd.release(); /* do not close */

```

Comment:
If you are not taking ownership of the file descriptor, wil::unique_fd doesn't make sense. You could either duplicate the file descriptor, or make it clear the class does not own the FD.

Reply:
Right, I will change to dup it.